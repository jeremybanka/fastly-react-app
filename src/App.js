// @flow

import * as React from "react"

import { AuthPage, FastlyPage } from "./pages"
import { Box, ThemeProvider, theme as cosmoTheme, getTheme } from "cosmo"
import { GlobalStyle, Navigation } from "./components"
import { Redirect, Route, Switch, useLocation } from "react-router-dom"

import { RecoilRoot } from "recoil"
import useLocalStorage from "./hooks/useLocalStorage"

function App(): React.Node {
  const location = useLocation()

  // get browser-defined preference for dark mode
  const prefersDarkMode =
    window.matchMedia &&
    window.matchMedia("(prefers-color-scheme: dark)").matches

  // store theme preference in local storage
  const [theme, setTheme] = useLocalStorage(
    "theme",
    prefersDarkMode ? "dark" : "light"
  )

  // e.g. 'light' â†’ 'fastlyLight'
  const brandedTheme = location.pathname.includes("fastly")
    ? `fastly${theme.replace(theme[0], theme[0].toUpperCase())}`
    : theme

  return (
    <RecoilRoot>
      <ThemeProvider theme={getTheme(cosmoTheme, brandedTheme)}>
        <GlobalStyle />
        <Navigation theme={theme} onThemeChange={setTheme} />
        <Box padding="lg">
          <Switch>
            <Redirect exact from="/" to="/fastly" />
            <Route path={"/fastly/:serviceId?"}>
              <FastlyPage />
            </Route>
            <Route path="/auth">
              <AuthPage />
            </Route>
          </Switch>
        </Box>
      </ThemeProvider>
    </RecoilRoot>
  )
}

export default App
