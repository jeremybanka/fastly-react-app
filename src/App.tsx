import { AuthPage, PostPages, TlsConfigurationsPages } from "./pages"
/* eslint-disable @typescript-eslint/ban-ts-comment */
// @ts-ignore
import { Box, ThemeProvider, theme as cosmoTheme, getTheme } from "cosmo"
import { GlobalStyle, Navigation } from "./components"
import { QueryClient, QueryClientProvider } from "react-query"
import { Redirect, Route, Switch, useLocation } from "react-router-dom"
import AuthProvider from "./components/AuthProvider"

import React from "react"
import { ReactQueryDevtools } from "react-query/devtools"
import { RecoilRoot } from "recoil"
import useLocalStorage from "./hooks/useLocalStorage"

const queryClient = new QueryClient()

export default function App(): React.ReactNode {
  const location = useLocation()

  // get browser-defined preference for dark mode
  const prefersDarkMode = window?.matchMedia(`(prefers-color-scheme: dark)`).matches

  // store theme preference in local storage
  const [theme, setTheme] = useLocalStorage(`theme`, prefersDarkMode ? `dark` : `light`)

  // e.g. 'light' â†’ 'fastlyLight'
  // const brandedTheme = 'light'
  const brandedTheme = location.pathname.includes(`fastly`)
    ? `fastly${theme.replace(theme[0], theme[0].toUpperCase())}`
    : theme

  return (
    <QueryClientProvider client={queryClient}>
      <RecoilRoot>
        <ThemeProvider theme={getTheme(cosmoTheme, brandedTheme)}>
          <AuthProvider>
            <GlobalStyle />
            <Navigation theme={theme} onThemeChange={setTheme} />
            <Box padding="lg">
              <Switch>
                <Redirect exact from="/" to="/posts" />
                <Route path="/tls-configurations/:id">
                  <TlsConfigurationsPages.View />
                </Route>
                <Route path="/tls-configurations">
                  <TlsConfigurationsPages.Index />
                </Route>
                <Route path="/posts/:id">
                  <PostPages.View />
                </Route>
                <Route path="/posts">
                  <PostPages.Index />
                </Route>
                <Route path="/auth/sign-in">
                  <AuthPage />
                </Route>
              </Switch>
            </Box>
            <ReactQueryDevtools initialIsOpen />
          </AuthProvider>
        </ThemeProvider>
      </RecoilRoot>
    </QueryClientProvider>
  )
}
