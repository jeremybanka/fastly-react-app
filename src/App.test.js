import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import App from "./App";

// This is an example test that just snapshots the <App />.
// Expectation is that this test would be removed in favour of more discreet
// component tests.

test("example snapshot test for <App />", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
