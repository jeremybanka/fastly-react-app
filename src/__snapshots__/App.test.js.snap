// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example snapshot test for <App /> 1`] = `
<DocumentFragment>
  <div
    class="sc-kfzAmx fGnnXi"
  >
    <div
      class="sc-fKFyDc gTMLx"
    >
      <a
        aria-current="page"
        class="sc-bBXqnf flaYrT isActive"
        href="/fastly"
      >
        Fastly
      </a>
      <a
        class="sc-bBXqnf flaYrT"
        href="/origins"
      >
        Origin Inspector
      </a>
      <a
        class="sc-bBXqnf flaYrT"
        href="/domains"
      >
        Domain Inspector
      </a>
      <a
        class="sc-bBXqnf flaYrT"
        href="/realtime"
      >
        Real-Time
      </a>
      <a
        class="sc-bBXqnf flaYrT"
        href="/sigsci"
      >
        Signal Sciences
      </a>
      <a
        class="sc-bBXqnf flaYrT"
        href="/auth"
      >
        Auth
      </a>
    </div>
    <div
      class="sc-dlfnbm eiGZdo"
    />
    <div
      class="sc-dlfnbm isWJdv"
    >
      <div
        class="sc-dlfnbm isWJdv"
      >
        <div
          class="sc-jrAGrp ipUvmh"
        >
          <div
            class="sc-dlfnbm isWJdv"
          >
            <label
              class="sc-bkzZxe iTwkIl"
              for="theme"
            >
              <input
                class="sc-fFubgz hHftiS"
                id="theme"
                name="theme"
                type="checkbox"
                value="dark"
              />
              <div
                class="sc-idOhPF cKYVLL"
              />
            </label>
          </div>
          <div
            class="sc-dlfnbm cMVmUG"
          >
            <div
              class="sc-dlfnbm isWJdv"
            >
              <label
                class="sc-hHftDr bkogVI"
                color="grays.6"
                font-size="xs"
                font-weight="semibold"
                for="theme"
                letter-spacing="sm"
              >
                Dark mode
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-dlfnbm ePuIJd"
  >
    <div
      class="sc-dlfnbm gxZzRO"
    >
      <span
        class="sc-hHftDr zPloS"
        font-weight="semibold"
      >
        ðŸ”‘ Auth credentials missing
      </span>
      <div
        class="sc-dlfnbm ikYQkK"
      />
      Are your auth credentials present and correct?
    </div>
  </div>
</DocumentFragment>
`;
